bindings/java/src/main/java/org/openucx/jucx/ucp/UcpContext.java

```diff
@@ -51,22 +52,32 @@ public UcpWorker newWorker(UcpWorkerParams params) {
     }
 
     /**
-     * Associates memory allocated/mapped region with the network resources.
+     * Associates memory allocated/mapped region with communication operations.
      * The network stack associated with an application context
      * can typically send and receive data from the mapped memory without
      * CPU intervention; some devices and associated network stacks
-     * require the memory to be mapped to send and receive data.
+     * require the memory to be registered to send and receive data.
      */
     public UcpMemory registerMemory(ByteBuffer buf) {
         if (!buf.isDirect()) {
-            throw new UcxException("Registered buffer must be direct.");
+            throw new UcxException("Registered buffer must be direct");
         }
-        return registerMemoryNative(getNativeId(), buf);
+        UcpMemory result = registerMemoryNative(getNativeId(), UcxUtils.getAddress(buf),
+            buf.remaining(), 0L);
+
+        result.setByteBufferReference(buf);
```

Comment:
can we set this reference from native code, so user cannot use setByteBufferReference()?

Reply:
this method is package-private:
>package-private (often just called package) means that other members of the same package have access to the item. package-private is the default access modifier and does not have a keyword, because package is used to specify the package for a class or interface. To declare package access for something, use no access mod

Reply:
So user can't use it, unless using java magic, to bypass access control.

Reply:
ok